### ISSUE TITLE
Support left padding with torch.nn.functional.pad

### ISSUE THREAD

**Issue by avery-blanchard** [torch-spyre/torch-spyre#1480]
### What feature would you like to be added?

Support for right padding with torch.nn.functional.pad was added in #1310. The decomposition for this operation should be extended to support left padding through offsets.

### Why is this feature needed?

The torch.nn.functional.pad operation supports left and right padding. This operation is used in granite 4.0 (see [here](https://github.com/torch-spyre/torch-spyre/blob/main/tests/resource/models/granite-4.0-h-tiny.yaml#L1509-L1525))

**Comment by avery-blanchard** [comment]
cc: @michihirohorie

**Comment by avery-blanchard** [comment]
#1561
